internal/relayer_keyring/relayer_keyring.go

```diff
@@ -0,0 +1,60 @@
+package relayer_keyring
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	neutronapp "github.com/neutron-org/neutron/app"
+)
+
+type passReader struct {
+	pass string
+	buf  *bytes.Buffer
+}
+
+func newPassReader(pass string) io.Reader {
+	return &passReader{
+		pass: pass,
+		buf:  new(bytes.Buffer),
+	}
+}
+
+func (r *passReader) Read(p []byte) (n int, err error) {
+	n, err = r.buf.Read(p)
+	if err == io.EOF || n == 0 {
+		r.buf.WriteString(r.pass + "\n")
+
+		n, err = r.buf.Read(p)
+	}
+
+	return n, err
+}
+
+func InitializeKeyring(keyringBackend, keyringPassword, homeDir, keyName, keySeed, hdPath string) (keyring.Keyring, string, error) {
+	passReader := newPassReader(keyringPassword)
+
+	keybase, err := keyring.New(neutronapp.Bech32MainPrefix, keyringBackend, homeDir, passReader)
```

Comment:
have you tried passing
```go
strings.NewReader(keyringPassword)
```
instead of this complicated `passReader`?

Reply:
It didn't work for some reason before… It probably was some other error, looks like it works now. Replaced it with `strings.NewReader `